Src/PCompiler/CompilerCore/Backend/Stately/StatelyCodeGenerator.cs

```diff
@@ -0,0 +1,127 @@
+            WriteSourcePrologue(context, source.Stream);
+            // write the top level declarations
+            foreach (var decl in globalScope.AllDecls)
+            {
+                WriteDecl(context, source.Stream, decl);
+            }
+
+            return source;
+        }
+
+        private void WriteDecl(CompilationContext context, StringWriter output, IPDecl decl)
+        {
+            string declName;
+            switch (decl)
+            {
+                case Machine machine:
+                    if (!machine.IsSpec)
+                    {
+                        WriteMachine(context, output, machine);
+                    }
+
+                    break;
+            }
+        }
+
+        private void WriteMachine(CompilationContext context, StringWriter output, Machine machine)
+        {
+
+            context.WriteLine(output, $"const {machine.Name} = createMachine<Context>({{");
+            context.WriteLine(output, $"id: \"{machine.Name}\",");
+            
+            //Code start state of machine.
+            context.WriteLine(output,$"initial: \"{machine.StartState.Name}\", ");
+            
+            //Code up the states in each machine.
+            context.WriteLine(output, "states: {");
+            foreach (State state in machine.States)
+            {
+                
+                context.WriteLine(output,$"{state.Name}: {{");
+                WriteState(context, output, state);
+                context.WriteLine(output, state.Equals(machine.States.Last()) ? "}" : "},");
+            }
+            context.WriteLine(output, "}");
+            context.WriteLine(output, "});");
+        }
+
+        private void WriteState(CompilationContext context, StringWriter output, State state)
+        {
+            //Entry function exists!
+            if (state.Entry != null) {
+                foreach (var s in state.Entry.Body.Statements) {
+                    if (s.GetType() == typeof(GotoStmt)) {
+                        var x = (GotoStmt)s;
+                        context.WriteLine(output, "always: [");
+                        context.WriteLine(output, $"{{target: '{x.State.Name}'}}");
+                        context.WriteLine(output, "]");
+                    }
+                }
+            }
+            var gotoStmts = new List<(String, String)>();
+            foreach (var pair in state.AllEventHandlers)
+            {
+                var handledEvent = pair.Key;
+                
+                //context.WriteLine(output, $"{pair.Value}");
+                switch (pair.Value)
+                {
+                    case EventGotoState goAct:
+                        gotoStmts.Add((goAct.Trigger.Name, goAct.Target.Name));
+                        break;
+                    case EventDoAction doAct:
+                        foreach (var stmt in doAct.Target.Body.Statements)
+                        {
+                            if (stmt.GetType() == typeof(GotoStmt))
```

Comment:
Let's mark a TODO here (for later, version right now is already good enough) that if the `stmt` is a function call, we should also explore the function body for goto statements as well (that too recursively since a function body can make more function calls.

Reply:
@suesther Lets fix this in this version itself.

Reply:
Okay, I added the TODO

Reply:
It is a very small change; instead of a todo, can we fix it in this PR itself?

Reply:
@aman-goel Do you have an example of what you mean so I can test out if my code works?

Reply:
@suesther Here is a simple example (check the event handler for Pong in state Ping_WaitPong):

```
event Ping: machine;
event Pong;
event Success;

machine Main {
    var pongId: machine;

    start state Ping_Init {
        entry {
      	    pongId = new PONG();
    	    raise Success;   	   
        }
        on Success goto Ping_SendPing;
    }

    state Ping_SendPing {
        entry {
    	    send pongId, Ping, this;
    	    raise Success;
	    }
        on Success goto Ping_WaitPong;
     }

     state Ping_WaitPong {
        on Pong do {
            jumpToDoneNonDeterministic();
            if ($) {
                goto Ping_SendPing;
            }
        }
     }

     fun jumpToDoneNonDeterministic() {
        if ($) {
            jumpToDone();
        }
     }

     fun jumpToDone() {
        goto Done;
     }

    state Done {
        ignore Pong;
    }
}

machine PONG {
    start state Pong_WaitPing {
        entry { }
        on Ping goto Pong_SendPong;
    }

    state Pong_SendPong {
	entry (payload: machine) {
	     send payload, Pong;
	     raise Success;		 	  
	}
        on Success goto Pong_WaitPing;
    }
}
```